src/components/ComposedModal/ComposedModal.tsx

```diff
@@ -0,0 +1,143 @@
+import classnames from 'classnames';
+import { forwardRef } from 'react';
+import { getCommonProps, noOp } from '../../utils';
+import Button from '../Button/Button';
+import { ButtonVariants } from '../Button/types';
+import { Divider } from '../Divider';
+import Modal, { ModalProps } from '../Modal/Modal';
+import { Text, TextVariants } from '../Text';
+
+export interface ModalButtonProps {
+  /**
+   * Callback click function for button
+   */
+  onClick?: () => void | unknown;
+  /**
+   * Button label text
+   */
+  buttonLabel?: string;
+}
+
+export interface ComposedModalProps extends Omit<ModalProps, 'onClose' | 'role' | 'style'> {
+  /**
+   * Title for Composed Modal
+   */
+  title: string;
+  /**
+   * The content of the modal.
+   */
+  children: React.ReactNode;
+  /**
+   * onClose handler for the modal
+   */
+  onClose?: () => void;
+  /**
+   * Optional proprerty center align all text in the component
+   */
+  centerAlignText?: boolean;
```

Comment:
do we really need the `centerAlignText` since the component takes a `className` prop and those decisions could be managed there?